src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/CustomerRegisterSuccess.php

```diff
@@ -57,11 +57,25 @@ public function execute(Varien_Event_Observer $observer)
         $response          = $accountController->getResponse();
         $params            = $this->getUtmParams();
         $params['token']   = $code;
-        $redirectUrl       = Mage::getUrl('shopgate-customer/customer_account/create', $params);
+
+        if ($this->getRequestHelper()->cookieIsSet(Shopgate_Cloudapi_Helper_Request::KEY_SGCLOUD_IS_CHECKOUT)) {
```

Comment:
See comment on top, please use a wrapper function instead of cookieIsSet with the constant as param